src/disjoint_set.jl

```diff
@@ -13,14 +13,17 @@
 #
 ############################################################
 
-type IntDisjointSets
+type IntDisjointSets{T<:Union{Void, Int}}
     parents::Vector{Int}
     ranks::Vector{Int}
+    sizes::Vector{T}
     ngroups::Int
 
     # creates a disjoint set comprised of n singletons
-    IntDisjointSets(n::Integer) = new(collect(1:n), zeros(Int, n), n)
+    IntDisjointSets{T}(n::Integer) where {T<:Int} = new(collect(1:n), zeros(Int, n), ones(Int, n), n)
+    IntDisjointSets{T}(n::Integer) where {T<:Void} = new(collect(1:n), zeros(Int, n), Vector{Void}(), n)
 end
+IntDisjointSets(n::Integer, store_size::Bool=false) = store_size ? IntDisjointSets{Int}(n) : IntDisjointSets{Void}(n)
```

Comment:
Not inferrable, but given that there are inferrable alternatives this isn't terrible. Perhaps a code comment explaining this would be helpful.